src/states/joining_peer.rs

```diff
@@ -295,49 +261,47 @@ impl Base for JoiningPeer {
                     }
                 }
             }
+            Variant::NodeApproval(gen_info) => {
```

Comment:
Again probably want to check the the src type and destination types are as expected.

Reply:
I don't think the check happen here because we ignore the result of as_section / as_node.

Reply:
Oops, you are right. Forgot the question marks :flushed:

Reply:
Maybe could at least specify the type of the _ in this case, that would avoid that issue.

Reply:
Have you thought about the suggestion to be explicit with the type `let _: type = ...?;` maybe clippy would complain but that would make it explicit that we don't want  a Result<..>